src/Sentry/Protocol/Span.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol
+{
+    // https://develop.sentry.dev/sdk/event-payloads/span
+    public class Span : ISpan, IJsonSerializable
+    {
+        public SentryId SpanId { get; }
+        public SentryId? ParentSpanId { get; }
+        public SentryId TraceId { get; set; }
+        public DateTimeOffset StartTimestamp { get; set; }
+        public DateTimeOffset EndTimestamp { get; set; }
+        public string Operation { get; }
+        public string? Description { get; set; }
+        public SpanStatus? Status { get; set; }
+        public bool IsSampled { get; set; }
+
+        private ConcurrentDictionary<string, string>? _tags;
+        public IReadOnlyDictionary<string, string> Tags => _tags ??= new ConcurrentDictionary<string, string>();
+
+        private ConcurrentDictionary<string, object>? _data;
+        public IReadOnlyDictionary<string, object> Data => _data ??= new ConcurrentDictionary<string, object>();
+
+        internal Span(SentryId? spanId = null, SentryId? parentSpanId = null, string operation = "unknown")
+        {
+            SpanId = spanId ?? SentryId.Create();
+            ParentSpanId = parentSpanId;
+            TraceId = SentryId.Create();
+            StartTimestamp = EndTimestamp = DateTimeOffset.Now;
```

Comment:
Do we run the risk of sending a 0 duration span when in fact we should have sent a open span (no end time?)

What's the reason to initialize the value like this?

Reply:
I didn't know spans can have no end time

Reply:
(Late to the party, commenting in case it is still useful -- cc @bruno-garcia @Tyrrrz )

The current behavior for "open spans" (no end time set) in other SDKs is such that they are filtered out when finishing/sending the transaction.

E.g.:
- https://github.com/getsentry/sentry-python/blob/b7816b0cc100a47082922b8dd3e058134ad75d7c/sentry_sdk/tracing.py#L537-L553
- https://github.com/getsentry/sentry-javascript/blob/a833f0ec0a186f0672a536ea8f7d3b129a71a4fe/packages/tracing/src/transaction.ts#L90-L105